lock-api/src/main/java/com/palantir/lock/client/LeasedLockToken.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client;
+
+import java.util.UUID;
+
+import javax.annotation.concurrent.GuardedBy;
+
+import com.google.common.base.Preconditions;
+import com.palantir.lock.v2.IdentifiedTime;
+import com.palantir.lock.v2.Lease;
+import com.palantir.lock.v2.LockToken;
+
+final class LeasedLockToken implements LockToken {
```

Comment:
I don't think that this class should be in this PR. It's unused, so it's hard for a reviewer to understand the context of how it should be used in the future.

Reply:
Generally, a PR should try to do one thing; which means that it can be considered in its entirety. IMO the purpose of this PR is to split out the TimelockRpcClient from the TimelockService (which enables a second PR to add the new endpoints and test them on the server and a third PR which starts using them on the client, as we discussed).